File: LeetCode/valid_anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size())
            return false;
        
        unordered_map<char, int> hash;
        
        for(char c : s)
            hash[c]++;
        
        for(char c : t) {
            if(hash[c] > 0)
                hash[c]--;
            else
                return false;
        }
        
        return true;
    }
};

File: LeetCode/valid_anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }

    hash := map[byte]int{}

    for i := range t {
        hash[t[i]]++
    }

    for i := range s {
        hash[s[i]]--

        if hash[s[i]] < 0 {
            return false
        }
    }

    return true
}
